core/src/main/java/com/spotify/ffwd/output/CoreOutputManager.java

```diff
@@ -181,6 +200,35 @@ public final Long getHLLPRefreshPeriodLimit() {
         } else {
             this.cardinalityLimit = null;
         }
+
+        if (!Strings.isNullOrEmpty(dynamicTagsFile)) {
+            final File file = new File(dynamicTagsFile);
+            if (file.exists() && file.canRead()) {
+                final int initialDelay =
+                    System.getenv().containsKey("IS_FFWD_CONFIGURATION_TEST") ? 3 : 60;
+                tagRefreshingThread = Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder().setNameFormat("tag-refreshing-thread").build());
+                tagRefreshingThread.scheduleWithFixedDelay(() -> refreshTagsFromFile(file.toPath()),
+                    initialDelay, 60, TimeUnit.SECONDS);
+            } else {
+                log.warn("Cannot read file at {}. Ignoring it.", dynamicTagsFile);
+            }
+        }
+    }
+
+    private void refreshTagsFromFile(final Path path) {
+        try (final Stream<String> lines = Files.lines(path)) {
+            lines.forEach(label -> {
+                final String[] labelParts = label.split("=", 2);
+                // Only update tag values if the tag keys are explicitly specified in normal configs
+                if (tags.get(labelParts[0]) != null) {
```

Comment:
ugh, just realized this doesn't work right now. We set a ffwd common tag called `stage` that takes it's value from the K8s label `tugboat.spotify.com/stage`. So in the K8s downward API file, it'll be `tugboat.spotify.com/stage=[canary|production]`. And `tugboat.spotify.com/stage` != `stage` here so the tag won't be included.

Our options are to add prefix-stripping logic here so that we can specify which prefixes to remove like `tugboat.spotify.com/` 🤮  Or we modify tugboat to use the label `stage` instead of `tugboat.spotify.com/stage` (with a migration period where we have both labels for a while with the same values). Thoughts?

Reply:
Yeah, it does look ugly. Adding some sort of label mapping into FFWD configuration also sounds like too complicated.
I like the latter as well 👍

Reply:
+1 the latter would be preferable

Reply:
Cool we will do that. Are there any docs I should update before this is merged?